### ISSUE TITLE
API call with localhost is blocked by CORS

### ISSUE THREAD

**Issue by happyharis** [thepracticaldev/dev.to#7878]
**Describe the bug**

Similar to [#5877](https://github.com/thepracticaldev/dev.to/issues/5877). Tried to do an API call through the Chrome browser, CORS error found.

**To Reproduce**

Steps to reproduce the behaviour:

1. I opened my website with localhost
2. I went to the Inspector/Console and typed the following:

```js
await fetch("https://dev.to/api/articles/me/", {
  "headers": {
    "api-key":"API_KEY"
  },
  "method":"GET",
  "mode":"cors"
});
```

3. Error returned:
Access to fetch at 'https://dev.to/api/articles/me/' from origin 'http://localhost:60121' has been blocked by CORS policy: Request header field api-key is not allowed by Access-Control-Allow-Headers in preflight response.

**Expected behavior**
To render 200 OK response

**Desktop (please complete the following information):**

- OS: MacOS Catalina
- Browser: Chrome
- Version: Version 81.0.4044.138 (Official Build) (64-bit)

**Comment by rhymes** [comment]
Hi @happyharis, only unauthenticated calls are supported by CORS, see https://dev.to/devteam/changelog-query-the-dev-api-from-a-javascript-frontend-329j

I'm closing this in favor of https://github.com/thepracticaldev/dev.to/issues/5877 and https://github.com/thepracticaldev/dev.to/issues/4195

Thank you!

**Comment by timbogdanov** [comment]
@rhymes How does that make any sense though?
Can we not make requests to the authenticated endpoints? what is the point of the api-key?

**Comment by rhymes** [comment]
Hi! @timbogdanov - the problem here is that allowing authenticated requests from frontends would lead to hardcoding for API keys in frontends which we want to avoid. I think this can be enabled once we have a better authentication option for clients
